js/index.js

```diff
@@ -2,41 +2,85 @@
 
 function updateSubtotal(product) {
   console.log('Calculating subtotal, yey!');
+  
+
+const price = product.querySelector('.price span');
+const quantity = product.querySelector('.quantity input')
+
+const priceValue = price.innerText
+
+subtotal = priceValue * quantity.value
+
+console.log(priceValue)
+console.log(subtotal)
+
+
+const finalPrice = product.querySelector('.subtotal span')
+finalPrice.innerText = subtotal.toFixed(2)
 
-  //... your code goes here
 }
 
-function calculateAll() {
-  // code in the following two lines is added just for testing purposes.
-  // it runs when only iteration 1 is completed. at later point, it can be removed.
-  const singleProduct = document.querySelector('.product');
-  updateSubtotal(singleProduct);
-  // end of test
 
+function calculateAll() {
+ 
   // ITERATION 2
-  //... your code goes here
+
+
+  const products = document.querySelectorAll('.product').forEach(product => {
+    updateSubtotal (product);
+
+  });
+
 
   // ITERATION 3
-  //... your code goes here
+  const eachTotal = document.querySelectorAll('.subtotal span');
+  let allTotal = 0;
+
+  for (let i = 0; i < eachTotal.length; i++) {
+    allTotal += parseInt(eachTotal[i].innerHTML);
+  }
+  document.querySelector('#total-value span').innerHTML = allTotal;
 }
 
+
 // ITERATION 4
 
 function removeProduct(event) {
   const target = event.currentTarget;
   console.log('The target in remove is:', target);
-  //... your code goes here
+  const productToRemove = event.currentTarget.parentNode.parentNode
+  // document.productToRemove('.cart').removeChild(productToRemove)
+  productToRemove.remove()
 }
```

Comment:
Good job for this one! What is missing here is to update the total price after removing products. You do that by just calling `calculateAll()` at the end of the function to recalculate the total.